build.go

```diff
@@ -0,0 +1,161 @@
+
+// EntryResolver defines the interface for picking the most relevant entry from
+// the Buildpack Plan entries.
+type EntryResolver interface {
+	Resolve(string, []packit.BuildpackPlanEntry, []interface{}) (packit.BuildpackPlanEntry, []packit.BuildpackPlanEntry)
+	MergeLayerTypes(string, []packit.BuildpackPlanEntry) (launch, build bool)
+}
+
+// DependencyManager defines the interface for picking the best matching
+// dependency and installing it.
+type DependencyManager interface {
+	Resolve(path, id, version, stack string) (postal.Dependency, error)
+	Install(dependency postal.Dependency, cnbPath, destPath string) error
+}
+
+// BuildPlanRefinery defines the interface for generating a BuildpackPlan Entry
+// containing the Bill-of-Materials of a given dependency.
+type BuildPlanRefinery interface {
+	BillOfMaterial(dependency postal.Dependency) packit.BuildpackPlan
+}
+
+// InstallProcess defines the interface for installing the pip dependency into a layer.
+type InstallProcess interface {
+	Execute(srcPath, targetLayerPath string) error
+}
+
+// Build will return a packit.BuildFunc that will be invoked during the build
+// phase of the buildpack lifecycle.
+//
+// Build will find the right pip dependency to install, install it in a
+// layer, and generate Bill-of-Materials. It also makes use of the checksum of
+// the dependency to reuse the layer when possible.
+func Build(installProcess InstallProcess, entries EntryResolver, dependencies DependencyManager, planRefinery BuildPlanRefinery, logs scribe.Emitter, clock chronos.Clock) packit.BuildFunc {
+	return func(context packit.BuildContext) (packit.BuildResult, error) {
+		logs.Title("%s %s", context.BuildpackInfo.Name, context.BuildpackInfo.Version)
+
+		logs.Process("Resolving Pip version")
+
+		entry, sortedEntries := entries.Resolve(Pip, context.Plan.Entries, Priorities)
+
+		logs.Candidates(sortedEntries)
+
+		version, _ := entry.Metadata["version"].(string)
+
+		dependency, err := dependencies.Resolve(filepath.Join(context.CNBPath, "buildpack.toml"), entry.Name, version, context.Stack)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		dependency.Name = "Pip"
+		logs.SelectedDependency(entry, dependency, clock.Now())
+
+		bom := planRefinery.BillOfMaterial(dependency)
+
+		pipLayer, err := context.Layers.Get(Pip)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		cachedSHA, ok := pipLayer.Metadata[DependencySHAKey].(string)
+		if ok && cachedSHA == dependency.SHA256 {
+			logs.Process("Reusing cached layer %s", pipLayer.Path)
+			return packit.BuildResult{
+				Plan:   bom,
+				Layers: []packit.Layer{pipLayer},
+			}, nil
+		}
+
+		pipLayer, err = pipLayer.Reset()
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		pipLayer.Launch, pipLayer.Build = entries.MergeLayerTypes(Pip, context.Plan.Entries)
+		pipLayer.Cache = pipLayer.Build
+
+		pipSrcDir, err := ioutil.TempDir("", "pip-source")
```

Comment:
Could you please write a comment why we are unloading the dependency to a tmpdir which is atypical of buildpacks?